workflow-templates/im-deploy-az-function-manually-to-dev-subn.yml

```diff
@@ -0,0 +1,191 @@
+# Prerequisites
+# 1 - Make sure the az secrets have been added to the environment
+# 2 - Make sure the scm restrictions include the prod github runners (previously it had just octopus)
+
+name: Manually deploy AZ Function by ref in Dev subscription
+on:
+  workflow_dispatch:
+    inputs:
+      branch-tag-sha:
+        description: Ref (branch/tag/sha) for the code that should be deployed to the Prod subscription.
+        required: true
+      environment:
+        description: The environment to Deploy To.
+        required: true
+
+env:
+  PAGERDUTY_SERVICE_IDS: '[ "${{secrets.PAGERDUTY_SERVICE_ID}}" ]' # TODO:  Add your PD Service IDs here, the svc id usually has a format of P0XXXXX
+  PAGERDUTY_WINDOW_IN_MIN: 30 # TODO:  Verify the length of your PD Maintenance Window
+  PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ github.event.inputs.environment }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+  DEPLOY_BOARD_NUM: '' # TODO: Add the automated deployment board number or remove if not using an automated deployment project board.
+  AZ_FUNC_NAME: '' # TODO:  set this to your azure function's name
+  AZ_SLOT_NAME: '' # TODO:  If using slots, set this to the name of the azure deployment slot
+  DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+  PROJECT_ROOT: '' # TODO:  set this to the folder of your function. For most projects this will be ./src/ProjectName
+  ARTIFACT_NAME: 'Published Function' #TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+
+jobs:
+  validate-input:
+    runs-on: [ubuntu-20.04]
+    steps:
+      # TODO:  Update these steps if you don't use all these environments in the Development Subscription
+      - name: Clean Environment Input
+        id: clean-env
+        uses: im-open/map-input-action@v1.0.0
+        with:
+          input: ${{ github.event.inputs.environment }}
+          input_map: '{ "Dev": ["dev", "d", "development"], "QA": ["qa", "q"], "Stage": ["stg", "s", "stage"]  }'
+
+      - name: Check if environment is Dev, QA or Stage
+        if: ${{ steps.clean-env.outputs.mapped_input != 'Dev' &&  steps.clean-env.outputs.mapped_input != 'QA' &&  steps.clean-env.outputs.mapped_input != 'Stage'}}
+        run: |
+          echo "::error::The environment must be Dev, QA or Stage";
+          exit 1
+
+  deploy-code:
+    needs: validate-input
+    runs-on: [self-hosted, ubuntu-20.04]
+    environment: ${{github.event.inputs.environment}}
```

Comment:
If it got to this point, it's one of the values (case might be different) so it doesn't matter which one you use.  I think I probably gravitate toward the known value because I've had issues in other scenarios setting environment based on a variable set within another job in the workflow.  It also means I have to set outputs on the other one.

If you have a strong preference I can change it.  I kind of like the idea of keeping it more obvious where the value comes from.

Reply:
Your reasoning is sound enough for me! Especially since this is a template so the ease of understanding is pretty darn important.